components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/deployers/config/ServiceProviderConfigDeployer.java

```diff
@@ -55,37 +55,51 @@ public ServiceProviderConfigDeployer(int tenantId, String id) {
         this.id = id;
     }
 
+    /**
+     * Used to retrieve Id.
+     *
+     * @return id
+     */
     public String getId() {
+
         return id;
     }
 
     /**
      * Deploy the configuration.
      *
-     * @param configurationFileStream
-     * @throws RemoteFetchCoreException
+     * @param configurationFileStream ConfigurationFileStream of cloned application
+     * @throws RemoteFetchCoreException RemoteFetchCoreException
+     * @throws IOException              IOException
      */
     @Override
     public void deploy(ConfigurationFileStream configurationFileStream) throws RemoteFetchCoreException, IOException {
 
         ServiceProvider previousServiceProvider = null;
-
-        Tenant tenant = this.getTenantDomain();
-        String userName = tenant.getAdminName();
-        startTenantFlow(tenant.getDomain(), userName);
+        String userName = this.getTenantAdminName();
```

Comment:
https://docs.oracle.com/javase/tutorial/java/javaOO/thiskey.html
https://stackoverflow.com/questions/2411270/when-should-i-use-this-in-a-class